src/uimports.cpp

```diff
@@ -106,6 +106,34 @@ static BOOL UPDATE_IMPORTS_XX(HANDLE hProcess,
         }
     }
 
+	///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+	//Added code
+	if (inh.IMPORT_DIRECTORY.VirtualAddress!=NULL && inh.IMPORT_DIRECTORY.Size==0)
+	{
+		//Don't worry about changing the PE file, because the load information of the original PE header has been saved and will be restored. The change here is just for the following code to work normally
+		PIMAGE_IMPORT_DESCRIPTOR pImageImport=(PIMAGE_IMPORT_DESCRIPTOR)(pbModule+inh.IMPORT_DIRECTORY.VirtualAddress);
+		IMAGE_IMPORT_DESCRIPTOR ImageImport;
+
+		if(!ReadProcessMemory(hProcess,pImageImport,&ImageImport,sizeof(ImageImport),NULL))
+		{
+			_Analysis_assume_(FALSE);
```

Comment:
This tells the compiler it will never hit this point, so the goto right after will probably be completely elided.

Reply:
![image](https://user-images.githubusercontent.com/20564099/92300137-0b760d00-ef8b-11ea-9a75-c5285afbeeb3.png)
from IDA decompiled code, the `goto `code can work fine.


fir

Reply:
Ah, confused it for __assume.